codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/RangeTraitValidationErrorMessage.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy
+
+import software.amazon.smithy.model.traits.RangeTrait
+
+fun RangeTrait.validationErrorMessage(): String {
```

Comment:
Should probably make this DRYer with `LengthTraitValidationErrorMessage.kt` if the protocol tests specify the messages coincide.

Reply:
The message are currently slightly different, what do you mean by

> if the protocol tests specify the messages coincide.

?

Reply:
There are protocol tests that assert that we reject malformed requests, and stipulate the format of the error message that we need to return in the response: https://github.com/awslabs/smithy-rs/blob/87e4766f9896663e4e00dc8d33b9cc7c49601ea0/codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/protocol/ServerProtocolTestGenerator.kt#L987-L988

When we implement everything correctly, you should see those tests start passing and hence be forced to remove them from `ExpectFail`.

The format of the error message is [here](https://github.com/awslabs/smithy/blob/a8fd6c537d3d1e0c0258741efceda3ea85a8c75b/smithy-aws-protocol-tests/model/restJson1/validation/malformed-range.smithy#L81-L82). It doesn't begin with "The integer value", but rather with "Value x at...".

My comment above was saying that we could extract the definition of `ending`, since it seems to be the same in both error messages.

Reply:
I've noticed that Smithy only provides tests for malformed bytes and floats. We should write our own for `short`s, `integer`s, `long`s and `double`s and contribute them upstream. Since you're only generating constrained integers, you won't be able to test that the implementation is correct unless you write tests for `integer`s in this PR too.